generate.py

```diff
@@ -27,6 +29,12 @@ def _str_to_bool(s):
                              'boolean, got {}'.format(s))
         return {'true': True, 'false': False}[s.lower()]
 
+    def _ensure_positive_float(f):
```

Comment:
Didn't you add this function as part of the feature? It has a very generic name but it's specific to the temperature parameter, that's why I mentioned this. I think `argparse` takes care of referring to the argument name when displaying an error, so I suggested leaving the "positive float" generic rather than adding a `_parse_temperature` function. The `_str_to_bool` is due for a refactor anyway though, so we could do both of them together later on, but I feel it's a but unnecessary.